### PR TITLE
fix: add missing fields to form

### PR DESCRIPTION
**Pull request by JordanGerada3**
## Description

- Added shortName (legal name) to form
- Added Company identifier to from
- Added translations EN, DE
- Removed unused 'BMW AG' hardcoding from transition files 

## Why

Necessary information that the user should verify before registering was not available in the form. 

## Issue

#1140 

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: src/assets/locales/de/registration.json [hunk 1]
```diff
@@ -4,12 +4,13 @@
     "companyHeading": "Firmendetails",
     "companyMessage": "Unten finden Sie die Registrierungsdaten Ihres Unternehmens, die Ihnen von Ihrem Onboarding-Anbieter mitgeteilt wurden. Bitte überprüfen Sie die Details und fahren Sie mit der Auswahl der Unternehmensrolle unten fort, bevor Sie Ihre Registrierung absenden.",
     "companyName": "Name der Firma",
-    "bmw": "BMW AG",
```
Comment (saadanzari) [inline src/assets/locales/de/registration.json:7]: is this deleted on purpose?

Comment (JordanGerada3) [reply src/assets/locales/de/registration.json:7]: Yes, it was removed from the form but left in the transition files unused.

File: src/components/pages/OSPConsent/CompanyDetails.tsx [hunk 2]
```diff
@@ -227,6 +228,12 @@ export const CompanyDetails = ({
       [t('osp.city'), companyDetails?.city ?? ''],
       [t('osp.region'), companyDetails?.region ?? ''],
       [t('osp.country'), companyDetails?.countryAlpha2Code ?? ''],
+      [
+        t('osp.companyId'),
+        companyDetails?.uniqueIds?.[0]
```
Comment (ss-nikunj) [inline src/components/pages/OSPConsent/CompanyDetails.tsx:233]: What if, If we dont have uniqueIds?

File: CHANGELOG.md [hunk 3]
```diff
@@ -61,6 +61,7 @@
   - fixed missing street number in form [#1081](https://github.com/eclipse-tractusx/portal-frontend/issues/1081)
   - filtered out roles not defined by OSP [#1114](https://github.com/eclipse-tractusx/portal-frontend/pull/1114)
   - fixed OSP consent form 400 submission error [#1102](https://github.com/eclipse-tractusx/portal-frontend/pull/1102/files)
+  - add missing fields to form legal name & company identifier[#1151](https://github.com/eclipse-tractusx/portal-frontend/pull/1151)
```
Comment (typecastcloud) [inline CHANGELOG.md:64]: Please do not add changelog to already released versions.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,10 +2,17 @@
...
+
+- **Onboarding Service Provider**
+  - add missing fields to form legal name & company identifier[#1151](https://github.com/eclipse-tractusx/portal-frontend/pull/1151)
+
 ## 2.3.0-RC2
 
 ### Bugfixes
 
+- **App Subscription Management**
+  - fixed 'read more' link by adding link [#1200](https://github.com/eclipse-tractusx/portal-frontend/pull/1200)
```
Comment (typecastcloud) [inline CHANGELOG.md:15]: Why are you adding this changelog entry?

Comment (JordanGerada3) [reply CHANGELOG.md]: Now removed.

**Pull request closed without merging**